core/src/main/java/org/apache/accumulo/core/client/rfile/RFileScanner.java

```diff
@@ -179,13 +180,14 @@ public void indexWeightChanged() {}
     }
 
     this.opts = opts;
+    ConfigurationCopy tableConf = new ConfigurationCopy(DefaultConfiguration.getInstance());
```

Comment:
It would be nice to void creating the config copy if opt.tableConfig is null. Maybe the following would do this :

```java
AccumuloConfiguration tableConf;
if(null != opts.tableConfig) {
  tableConf = new ConfigurationCopy(DefaultConfiguration.getInstance());
  opts.tableConfig.forEach(tableConf::set);
} else {
  tableConf = DefaultConfiguration.getInstance();
}
```